### PR TITLE
[Hub Generated] Review request for Microsoft.Synapse to add version preview/2021-04-01-preview

### PR DESCRIPTION
**Pull request by amitelran**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/amitelran/azure-rest-api-specs/kustpool-2021-04-01-preview...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by amitelran** [comment]
@ravbhatnagar @josefree 
Hi guys,
We're almost complete with the PR and your assistance will be truly helpful for completing it as soon as possible:

- Gaurav: 
Encountered validation failures while removing the "location". 
Currently removed only _LocationNameParameter_ as can be seen in this commit:
https://github.com/Azure/azure-rest-api-specs/pull/13484/commits/30d3cd815204abe9cb2c4876c39d2048e457b46c

- Josephine:
It seems that the LintDiff failed check is failing over an already existing code we didn't modify.
Can you assist with that?

- ApiDocPreview failed check:
Can it be ignored?

**Comment by jianyexi** [comment]
@amitelran  For LintDiff error, the error message has a mistake , actually the missing Api is 
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/privateLinkHubs/{privateLinkHubName}/privateEndpointConnections/{privateEndpointConnectionName}

**Comment by zhenglaizhang** [comment]
Hi @amitelran you can ignore the ApiDocPreview issue, it's due to the autorest OOM issue, I will try to loop docs team to fix the issue.

File: specification/synapse/resource-manager/readme.md [hunk 1]
```diff
@@ -28,16 +28,39 @@ These are the global settings for the Azure Synapse Analytics API.
...
+  - Microsoft.Synapse/preview/2021-04-01-preview/library.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/operations.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/privateEndpointConnections.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/privateLinkResources.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/privatelinkhub.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/sqlPool.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/sqlServer.json
+  - Microsoft.Synapse/preview/2021-04-01-preview/workspace.json
+```
 ### Tag: package-2021-03
```
Comment (idear1203) [inline specification/synapse/resource-manager/readme.md:59]: Nice to add a space between this two lines

Comment (amitelran) [reply specification/synapse/resource-manager/readme.md:60]: Line added

**Comment by amitelran** [comment]
@ravbhatnagar @akning-ms 
Following a request from the Synapse team, I have added a "DoNotMerge" label to avoid PR merge if approved in the meantime.
The Synapse team are looking into a new process regarding the Swagger and asked to delay the merge.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsCreateOrUpdate.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
+{
```
Comment (praveenmathamsetty) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsCreateOrUpdate.json:1]: Workspace Uid is exists as part of payload

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-123456789098",
```
Comment (praveenmathamsetty) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json:3]: Workspace Uid is exists as part of payload

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "resourceGroupName": "kustorptest",
+    "workspaceName": "kustorptest",
+    "kustoPoolName": "kustoclusterrptest4",
+    "api-version": "2021-04-01-preview",
+    "parameters": {
+      "properties": {
+        "workspaceUid": "11111111-2222-3333-444444444444"
+      }
+    }
```
Comment (praveenmathamsetty) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json:12]: Delete don't have body

**Review by ruowan — APPROVED** [review]

**Comment by idear1203** [comment]
>   clear-output-folder: true

For preview API version, we would like to include SQL DW GEN3 related APIs in the C# SDK too. Could you please help add the following lines below:

```
batch:
 - tag: package-sqlGen3-2020-04-01-preview
 - tag: package-preview-2021-04
```

---
Refers to: specification/synapse/resource-manager/readme.md:263 in 3a96932. [](commit_id = 3a969322c66b7b4365da76d9edac17a0708a1844, deletion_comment = False)

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json (hunk 4 above)
Comment (amitelran) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolsDelete.json]: Removed

**Comment by amitelran** [comment]
PR is approved.
Can merge the PR?
The failed validation can be ignored according to Zhenglai Zhang.

**Comment by idear1203** [comment]
@amitelran , could you please help update the configuration(REARME.md) of C# section as below. Otherwise we will have to open another PR for this which will be time consuming...

```
## C#
These settings apply only when `--csharp` is specified on the command line.
Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.
``` yaml $(csharp)
csharp:
  license-header: MICROSOFT_MIT_NO_VERSION
  namespace: Microsoft.Azure.Management.Synapse
  output-folder: $(csharp-sdks-folder)/synapse/Microsoft.Azure.Management.Synapse/src/Generated
  clear-output-folder: true
batch:
 - tag: package-sqlGen3-2020-04-01-preview
 - tag: package-preview-2021-04
```

**Comment by amitelran** [comment]
> @amitelran , could you please help update the configuration([REARME.md](https://github.com/amitelran/azure-rest-api-specs/blob/kustpool-2021-04-01-preview/specification/synapse/resource-manager/readme.md)) of C# section as below. Otherwise we will have to open another PR for this which will be time consuming...
> 
> ```
> ## C#
> These settings apply only when `--csharp` is specified on the command line.
> Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.
> ``` yaml $(csharp)
> csharp:
>   license-header: MICROSOFT_MIT_NO_VERSION
>   namespace: Microsoft.Azure.Management.Synapse
>   output-folder: $(csharp-sdks-folder)/synapse/Microsoft.Azure.Management.Synapse/src/Generated
>   clear-output-folder: true
> batch:
>  - tag: package-sqlGen3-2020-04-01-preview
>  - tag: package-preview-2021-04
> ```
> ```

**Comment by amitelran** [comment]
@idear1203 
Though I understand the will to avoid going through PR stages, your requested changes does not belong to our PR.
Regardless, your requested changes fail validation, so I suggest sticking with the usual way of publishing a new PR for your changes.
We try completing this PR for a long time, and such comments block us from doing so.
Thanks.

**Comment by ruowan** [comment]
The model validation failed. Could you pls fix the model validation?

**Pull request closed without merging**